examples/interactive_asr/vad.py

```diff
@@ -0,0 +1,254 @@
+"""
+Following `a simple but efficient real-time voice activity detection
+algorithm <https://www.eurasip.org/Proceedings/Eusipco/Eusipco2009/contents/papers/1569192958.pdf>`__.
+
+"""
+
+import math
+
+import pyaudio
+import numpy as np
+import librosa
+
+import torch
+import torchaudio
+
+from six.moves import queue
+from collections import deque
+
+######################################################################
+# There are three criteria to decide if a frame contains speech: energy,
+# most dominant frequency, and spectral flatness. If any two of those are
+# higher than a minimum plus a threshold, then the frame contains speech.
+# In the offline case, the list of frames is postprocessed to remove too
+# short silence and speech sequences. In the online case here, inertia is
+# added before switching from speech to silence or vice versa.
+#
+
+
+def compute_spectral_flatness(frame):
+    EPSILON = 0.01
+    n = frame.nonzero().size(0)
+    geometric_mean = torch.exp((EPSILON + frame).log().mean(-1)) - EPSILON
+    arithmetic_mean = frame.mean(-1)
+    return -10 * torch.log10(EPSILON + geometric_mean/arithmetic_mean)
+
+
+def compute_energy(frame):
+    return frame.pow(2).sum(-1)
+
+
+class VoiceActivityDetection(object):
+
+    def __init__(self):
+
+        self.num_init_frames = 30
```

Comment:
it would be nicer to expose `num_init_frames`, `ignore_silent_count`, `ignore_speech_count`, `energy_prim_thresh`, `frequency_prim_thresh`, `spectral_flatness_prim_thresh`, `ignore_silent_count`, `ignore_speech_count` through the `__init__` parameters.